### ISSUE TITLE
Hyperparameter Tuning, Optimization & EDA

### ISSUE THREAD

**Issue by Saswat-Subhadarsi** [Saswat-Subhadarsi/Asclepius#4]
Improve model performance by adjusting learning rate, embedding dimensions, batch size, etc
